### PR TITLE
test(Migration to RTL): Refactor ChartTable.test.tsx from Enzyme to RTL

### PR DESCRIPTION
**Pull request by rtexelm**
### SUMMARY

Replace enzyme with RTL functions. Refactor the tests accordingly. Added a test for the `mine` tab.

### TESTING INSTRUCTIONS

`npm test ChartTable.test.tsx`

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset-frontend/src/features/home/ChartTable.test.tsx [hunk 1]
```diff
@@ -68,64 +69,53 @@ describe('ChartTable', () => {
...
   });
 
   it('fetches chart favorites and renders chart cards', async () => {
-    act(() => {
-      const handler = wrapper.find('[role="tab"] a').at(0).prop('onClick');
-      if (handler) {
-        handler({} as any);
-      }
+    await renderChartTable(mockedProps);
+    fireEvent.click(screen.getByText(/favorite/i));
```
Comment (geido) [inline superset-frontend/src/features/home/ChartTable.test.tsx:102]: Let's use `userEvent` instead

Comment (rtexelm) [reply superset-frontend/src/features/home/ChartTable.test.tsx:102]: @justinpark curious what should be the go-to usage, `fireEvent` or `userEvent`, [as you mentioned](https://github.com/apache/superset/pull/27422/files/c9930915bd2be345a4af772953f2995b5b4545f4#r1516655143) we should save `userEvent` for special cases but this seems to go against the [RTL wiki](https://github.com/apache/superset/wiki/Testing-Guidelines-and-Best-Practices#react-testing-library-rtl) and [docs](https://testing-library.com/docs/user-event/intro#differences-from-fireevent)

**Review by justinpark — APPROVED** [review]
LGTM

**Review by geido — APPROVED** [review]

File: superset-frontend/src/features/home/ChartTable.test.tsx (hunk 1 above)
Comment (rtexelm) [reply superset-frontend/src/features/home/ChartTable.test.tsx]: @justinpark sent [this article ](https://stackoverflow.com/questions/61080116/ant-design-v4-breaks-react-testing-library-tests-for-select-and-autocomplete). Switching to `userEvent`

**Pull request merged**